ci-operator/step-registry/proxy/whitelist/gcp/proxy-whitelist-gcp-commands.sh

```diff
@@ -6,9 +6,15 @@ set -o pipefail
 
 # https://docs.openshift.com/container-platform/4.17/installing/install_config/configuring-firewall.html#configuring-firewall
 
-cat <<EOF > ${SHARED_DIR}/proxy_whitelist.txt
+if [[ "${ENABLE_GCP_CUSTOM_ENDPOINT}" == "yes" ]]; then
+    cat <<EOF > ${SHARED_DIR}/proxy_whitelist.txt
+.p.googleapis.com
```

Comment:
@jianli-wei according to the [docs](https://docs.redhat.com/en/documentation/openshift_container_platform/4.19/html/installing_on_gcp/installing-restricted-networks-gcp-installer-provisioned#installation-about-restricted-networks_installing-restricted-networks-gcp-installer-provisioned), "If you choose to perform a restricted network installation on a cloud platform, you still require access to its cloud APIs". In this case we need to whitelist `oauth2.googleapis.com`. It does not follow the `p.googleapis.com` format, but this is used for authentication and should not be changed unless you have an entire custom solution for authentication which would require massive changes to all parts of the openshift system. 

/cc @patrickdillon